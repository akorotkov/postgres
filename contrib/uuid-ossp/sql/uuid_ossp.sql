CREATE EXTENSION "uuid-ossp";

SELECT uuid_nil();
SELECT uuid_ns_dns();
SELECT uuid_ns_url();
SELECT uuid_ns_oid();
SELECT uuid_ns_x500();

SELECT uuid_generate_v1() < uuid_generate_v1();
SELECT uuid_generate_v1() < uuid_generate_v1mc();

SELECT substr(uuid_generate_v1()::text, 25) = substr(uuid_generate_v1()::text, 25);
SELECT substr(uuid_generate_v1()::text, 25) <> substr(uuid_generate_v1mc()::text, 25);
SELECT substr(uuid_generate_v1mc()::text, 25) <> substr(uuid_generate_v1mc()::text, 25);

SELECT ('x' || substr(uuid_generate_v1mc()::text, 25, 2))::bit(8) & '00000011';

SELECT uuid_generate_v3(uuid_ns_dns(), 'www.widgets.com');
SELECT uuid_generate_v5(uuid_ns_dns(), 'www.widgets.com');

SELECT uuid_generate_v4()::text ~ '^[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}$';
SELECT uuid_generate_v4() <> uuid_generate_v4();
