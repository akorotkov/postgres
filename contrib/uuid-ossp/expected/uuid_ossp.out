CREATE EXTENSION "uuid-ossp";
SELECT uuid_nil();
               uuid_nil               
--------------------------------------
 00000000-0000-0000-0000-000000000000
(1 row)

SELECT uuid_ns_dns();
             uuid_ns_dns              
--------------------------------------
 6ba7b810-9dad-11d1-80b4-00c04fd430c8
(1 row)

SELECT uuid_ns_url();
             uuid_ns_url              
--------------------------------------
 6ba7b811-9dad-11d1-80b4-00c04fd430c8
(1 row)

SELECT uuid_ns_oid();
             uuid_ns_oid              
--------------------------------------
 6ba7b812-9dad-11d1-80b4-00c04fd430c8
(1 row)

SELECT uuid_ns_x500();
             uuid_ns_x500             
--------------------------------------
 6ba7b814-9dad-11d1-80b4-00c04fd430c8
(1 row)

SELECT uuid_generate_v1() < uuid_generate_v1();
 ?column? 
----------
 t
(1 row)

SELECT uuid_generate_v1() < uuid_generate_v1mc();
 ?column? 
----------
 t
(1 row)

SELECT substr(uuid_generate_v1()::text, 25) = substr(uuid_generate_v1()::text, 25);
 ?column? 
----------
 t
(1 row)

SELECT substr(uuid_generate_v1()::text, 25) <> substr(uuid_generate_v1mc()::text, 25);
 ?column? 
----------
 t
(1 row)

SELECT substr(uuid_generate_v1mc()::text, 25) <> substr(uuid_generate_v1mc()::text, 25);
 ?column? 
----------
 t
(1 row)

SELECT ('x' || substr(uuid_generate_v1mc()::text, 25, 2))::bit(8) & '00000011';
 ?column? 
----------
 00000011
(1 row)

SELECT uuid_generate_v3(uuid_ns_dns(), 'www.widgets.com');
           uuid_generate_v3           
--------------------------------------
 3d813cbb-47fb-32ba-91df-831e1593ac29
(1 row)

SELECT uuid_generate_v5(uuid_ns_dns(), 'www.widgets.com');
           uuid_generate_v5           
--------------------------------------
 21f7f8de-8051-5b89-8680-0195ef798b6a
(1 row)

SELECT uuid_generate_v4()::text ~ '^[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}$';
 ?column? 
----------
 t
(1 row)

SELECT uuid_generate_v4() <> uuid_generate_v4();
 ?column? 
----------
 t
(1 row)

